alert-common/src/main/java/com/synopsys/integration/alert/common/action/UploadEndpointManager.java

```diff
@@ -90,83 +88,74 @@ public void unRegisterTarget(String targetKey) throws AlertException {
         uploadTargets.remove(targetKey);
     }
 
-    public ResponseEntity<String> performUpload(String targetKey, Resource fileResource) {
+    public void performUpload(String targetKey, Resource fileResource) {
         if (!containsTarget(targetKey)) {
-            return new ResponseEntity(NO_UPLOAD_FUNCTIONALITY_REGISTERED, HttpStatus.NOT_IMPLEMENTED);
+            throwNotImplementedException();
         }
 
         UploadTarget target = uploadTargets.get(targetKey);
         if (!authorizationManager.hasUploadWritePermission(target.getContext().name(), target.getDescriptorKey().getUniversalKey())) {
-            return responseFactory.createForbiddenResponse();
+            throw ResponseFactory.createForbiddenException();
         }
```

Comment:
I think these if statements can also be made into a throw method that is passed a function for which check to perform.

Reply:
It looks like they're checking different permissions, so I'm gonna leave them.

Reply:
Sure, but I think something like this would look cleaner
`private blah(BiFunction<Boolean, String, String> checkPermission, Target target) {
	if (!checkPermission.apply(target.getContext().name(), target.getDescriptorKey().getUniversalKey())) {
            throw ResponseFactory.createForbiddenException();
        }
}`
then just calling
`blah(authorizationManager::hasUploadWritePermission, target);`

Reply:
wow, didn't keep any formatting

Reply:
Ok, fine. I was just being lazy.